tools/cell_census_builder/validate.py

```diff
@@ -74,58 +78,85 @@ def validate_all_soma_objects_exist(soma_path: str, experiment_builders: List[Ex
         )
         assert sorted(census_info[CENSUS_SUMMARY_NAME].keys()) == sorted(["label", "value", "soma_joinid"])
 
+        # verify required dataset fields are set
+        census_info[CENSUS_DATASETS_NAME]
+        df: pd.DataFrame = census_info[CENSUS_DATASETS_NAME].read().concat().to_pandas()
+        assert not all(df["collection_id"].isin([""]))
+        assert not all(df["collection_name"].isin([""]))
+        assert not all(df["collection_doi"].isin([""]))
+        assert not all(df["dataset_title"].isin([""]))
+
         # there should be an experiment for each builder
         census_data = census[CENSUS_DATA_NAME]
-        for eb in experiment_builders:
-            assert soma.Experiment.exists(census_data[eb.name].uri)
-            assert census_data[eb.name].soma_type == "SOMAExperiment"
-
-            e = census_data[eb.name]
-            assert soma.DataFrame.exists(e.obs.uri)
-            assert e.obs.soma_type == "SOMADataFrame"
-            assert soma.Collection.exists(e.ms.uri)
-            assert e.ms.soma_type == "SOMACollection"
-
-            # there should be a single measurement called 'RNA'
-            assert soma.Measurement.exists(e.ms["RNA"].uri)
-            assert e.ms["RNA"].soma_type == "SOMAMeasurement"
-
-            # The measurement should contain all X layers where n_obs > 0 (existence checked elsewhere)
-            rna = e.ms["RNA"]
-            assert soma.DataFrame.exists(rna["var"].uri)
-            assert rna["var"].soma_type == "SOMADataFrame"
-            assert soma.Collection.exists(rna["X"].uri)
-            assert rna["X"].soma_type == "SOMACollection"
-            for lyr in CENSUS_X_LAYERS:
-                # layers only exist if there are cells in the measurement
-                if lyr in rna.X:
-                    assert soma.SparseNDArray.exists(rna.X[lyr].uri)
-                    assert rna.X[lyr].soma_type == "SOMASparseNDArray"
-
-            # and a dataset presence matrix
-            # dataset presence only exists if there are cells in the measurement
-            if eb.n_obs > 0:
-                assert soma.SparseNDArray.exists(rna[FEATURE_DATASET_PRESENCE_MATRIX_NAME].uri)
-                assert rna[FEATURE_DATASET_PRESENCE_MATRIX_NAME].soma_type == "SOMASparseNDArray"
-                assert sum([c.non_zero_length for c in rna["feature_dataset_presence_matrix"].read().coos()]) > 0
-                # TODO(atolopko): validate 1) shape, 2) joinids exist in datsets and var
-
+        with tiledb.Group(census_data.uri) as census_data_group:
+            for eb in experiment_builders:
+                assert soma.Experiment.exists(census_data[eb.name].uri)
+                assert census_data[eb.name].soma_type == "SOMAExperiment"
+                assert census_data_group.is_relative(eb.name), error_tiledb_group
+
+                e = census_data[eb.name]
+                with tiledb.Group(e.uri) as e_group:
```

Comment:
I think this code might be cleaner if you factored out the relative check into a stand-alone function, which would isolate all of the TileDB-specific code, and partition the "concern" being tested.  For example, reading this, it is a bit hard to figure out if you are testing the `relative` attribute on all objects, or have missed a few.

you could simply do a recursive, top-down walk of the root Collection, and check every child below the root for this flag.  Much less code, and (I think) easier to understand.